Description: Allow empty port in config
 MySQL 5.7 no longer allows empty values for non-string options in
 configuration files. The package dbconfig-common assumes this is ok.
 This patch is will prevent errors in packages using dbconfig until the
 bug is fixed.
Author: Lars Tangvald <lars.tangvald@oracle.com>
Origin: upstream
Bug: http://bugs.mysql.com/bug.php?id=80903
Bug-Ubuntu: https://launchpad.net/bugs/1563274
Forwarded: no
Last-Update: 2016-03-30

--- mysql-5.7-5.7.11.orig/mysys_ssl/my_getopt.cc
+++ mysql-5.7-5.7.11/mysys_ssl/my_getopt.cc
@@ -741,10 +741,19 @@ static int setval(const struct my_option
       )
      )
   {
-    my_getopt_error_reporter(ERROR_LEVEL,
+    if (strncmp(opts->name, "port", 10) == 0)
+    {
+        my_getopt_error_reporter(WARNING_LEVEL,
+                             "%s: Empty value for '%s' specified. Will throw an error in future versions",
+                             my_progname, opts->name);
+    }
+    else
+    {
+        my_getopt_error_reporter(ERROR_LEVEL,
                              "%s: Empty value for '%s' specified",
                              my_progname, opts->name);
-    return EXIT_ARGUMENT_REQUIRED;
+        return EXIT_ARGUMENT_REQUIRED;
+    }
   }
 
   if (value)
